include/mbedtls/runtime.h

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Constants and prototype of alternative runtime detection function for core
+ *  CPU feature sets.
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 ( the "License" ); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#if !defined(MBEDTLS_RUNTIME_H)
+#define MBEDTLS_RUNTIME_H
+
+#include <stdbool.h>
+
+#include "mbedtls/build_info.h"
+
+/* Build-time architecture detection utils. */
+#if !defined(MBEDTLS_ARCH_IS_ARM64) && \
```

Comment:
These macros should be used everywhere - there are lots of places that are still doing this sort of check by looking for `__aarch64__` etc (e.g., `bn_mul.h` - but there are lots of other examples)

Reply:
I will re-write this PR. 

What's the best place for these macros? I think `mbedtls/build_info.h` or `mbedtls/platform.h` might be better. But I do not think they should be exposed to user, so I will put them into `library/common.h` . 

Another thing is about the config checks, I prefer follow bellow ruler.
- Compiler check and CPU modifiers should be put in accelerator module.
- Config option checks should be put in `runtime.h`. 
- `MBEDTLS_RUNTIME_HAVE_CODE` will be enabled in `runtime.h` 
   - Any algorithm module has there own option for runtime detection. like `MBEDTLS_AES_RUNTIME_HAVE_CODE`
   - if any algorithm module enable runtime detection, `MBEDTLS_RUNTIME_HAVE_CODE` will be enabled.